### PR TITLE
replacing default protocol HTTP with HTTPS in WalledGardenInternetObs…

### LINKED ISSUE #323: Update protocol to https for DEFAULT_HOST in WalledGardenInternetObservingStrategy
See issue mentioned here: https://github.com/pwittchen/ReactiveNetwork/issues/299#issuecomment-462904722

Line in the code: https://github.com/pwittchen/ReactiveNetwork/blob/40b725013b8412289f56fec2601c4cab3f0fc76f/library/src/main/java/com/github/pwittchen/reactivenetwork/library/rx2/internet/observing/strategy/WalledGardenInternetObservingStrategy.java#L41

It needs to be verified and tested before pushing.

### PR DESCRIPTION
**Pull request by Kisty**
This PR introduces the following update from RxJava2.x branch to fix #323

### CODE REVIEW

**Comment by Kisty** [comment]
Hmm, not sure what to do about this.

File: library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java [hunk 1]
```diff
@@ -26,21 +45,36 @@
...
+  @Spy public ErrorHandler errorHandler = new ErrorHandler() {
+    @Override
+    public void handleError(Exception exception, String message) {
+      // do nothing
+    }
+  };
+
+  @Before
+  public void setUp() throws Exception {
+    Mockito.doNothing().when(errorHandler).handleError(ArgumentMatchers.any(Exception.class), ArgumentMatchers.anyString());
```
Comment (pwittchen) [inline library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java:63]: Please, use static import to exclude `Mockito` class name. It should look like: `doNothing().when(...)`. All tests in this project use that convention.

File: library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java [hunk 2]
```diff
@@ -26,21 +45,36 @@
...
 
   private String getHost() {
     return strategy.getDefaultPingHost();
   }
 
   @Test public void shouldBeConnectedToTheInternet() {
     // given
-    when(strategy.isConnected(getHost(), PORT, TIMEOUT_IN_MS, errorHandler)).thenReturn(true);
+    String host = getHost();
+    Mockito.doReturn(true).when(strategy).isConnected(host, PORT, TIMEOUT_IN_MS, errorHandler);
```
Comment (pwittchen) [inline library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java:73]: Please, use static import to exclude `Mockito` class name. It should look like: `doReturn(true).when(...)`. All tests in this project use that convention.

File: library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java [hunk 3]
```diff
@@ -51,11 +85,12 @@ private String getHost() {
 
   @Test public void shouldNotBeConnectedToTheInternet() {
     // given
-    when(strategy.isConnected(getHost(), PORT, TIMEOUT_IN_MS, errorHandler)).thenReturn(false);
+    String host = getHost();
+    Mockito.doReturn(false).when(strategy).isConnected(host, PORT, TIMEOUT_IN_MS, errorHandler);
```
Comment (pwittchen) [inline library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java:89]: Please, use static import to exclude `Mockito` class name. It should look like: `doReturn(false).when(...)`. All tests in this project use that convention.

File: library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java [hunk 4]
```diff
@@ -81,58 +116,80 @@ private String getHost() {
...
     // then
-    assertThat(transformedHost).isEqualTo(expectedHost);
+    assertThat(transformedHost).isEqualTo(HOST_WITH_HTTPS);
   }
 
   @Test public void shouldAdjustHostWhileCheckingConnectivity() {
     // given
     final String host = getHost();
-    when(strategy.isConnected(host, PORT, TIMEOUT_IN_MS, errorHandler)).thenReturn(true);
+    Mockito.doReturn(true).when(strategy).isConnected(host, PORT, TIMEOUT_IN_MS, errorHandler);
```
Comment (pwittchen) [inline library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java:192]: Please, use static import to exclude `Mockito` class name. It should look like: `doReturn(true).when(...)`. All tests in this project use that convention.

**Review by pwittchen — CHANGES_REQUESTED** [review]
Hi,

Thanks for your PR! I've added a few small comments in the review. When you will be done with fixes, please squash your commits into one to keep the git log of the project clean.

Don't worry about Travis. It's configured to fail, when test coverage decreases. I tried to configure it to fail, when difference of test coverage between PR and target branch is huge (e.g. 20%), but it looks like it didn't work. In case of 1% difference or less we can ignore it.

File: library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java (hunk 1 above)
Comment (Kisty) [reply library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java:63]: I tried that, but PMD check failed due to too many static imports :( (>4 static imports). Same for the other review issues. Any ideas how to move forward, or you happy to ignore?

Comment (pwittchen) [reply library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java:63]: Ok, then we can ignore it for now within this PR and optionally update PMD config later.

Comment (Kisty) [reply library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java:63]: Cool. Anything else you need from me?

Comment (pwittchen) [reply library/src/test/java/com/github/pwittchen/reactivenetwork/library/internet/observing/strategy/WalledGardenInternetObservingStrategyTest.java:63]: No. Just squash you commits into one, please. :-)

**Comment by Kisty** [comment]
Squished!

**Comment by Kisty** [comment]
Morning! Are you able to cook up a release for RxJava 1.x, please? Trying to use JitPack to make a custom build but having trouble, though it looks like it's working now. A proper release would be better.

Incidentally, would it be better to use the RxJava2 implementation? Can RxJava 1 & 2 coexist in same app?

I also tried adding as a submodule but can't due to `rootProject.ext` calls which the `ext` fields aren't at my root. I could put them manually, but that would be hacky. Unless you've solved this for RxJava 2.x branch, this might be worth looking in to.

**Comment by pwittchen** [comment]
Hi,

I'll try to release new version today in the evening. It's my side project and I'm not able to do it earlier.

> Can RxJava 1 & 2 coexist in same app?

Theoretically they can, but it's not good idea because you can easily mix imports and classes from different releases/packages what can cause errors. You can also use interop libraries to convert RxJava1 code into RxJava2 code dynamically. Maybe you can keep both of them during the migration phase, but it shouldn't be long-term solution.

**Comment by Kisty** [comment]
Thanks, man. Appreciate that and thanks for maintaining this great project!

**Comment by pwittchen** [comment]
PR merged.
Update will be available in version `0.13.0` -> #377

**Comment by Kisty** [comment]
Thanks a lot, man!

**Pull request merged**